docs/src/views/Layout.tsx

```diff
@@ -0,0 +1,212 @@
+import * as React from 'react'
+import { Button, Flex, Header, Icon, Segment, Divider } from '@stardust-ui/react'
+
+import CodeSnippet from '../components/CodeSnippet'
+import ExampleSnippet from '../components/ExampleSnippet'
+import DocPage from '../components/DocPage'
+import GuidesNavigationFooter from '../components/GuidesNavigationFooter'
+
+const code = value => <code>{value}</code>
+
+const link = (content, href, isExternal = false) => (
+  <a href={href} {...isExternal && { target: 'blank' }}>
+    {content} {isExternal ? <Icon name="external" size="small" link fitted /> : ''}
+  </a>
+)
+
+const links = {
+  flex: link('Flex', '/components/flex'),
+  flexItem: link('Flex.Item', '/components/flex'),
+  grid: link('Grid', '/components/grid'),
+  segment: link('Segment', '/components/segment'),
+}
+
+export default () => (
+  <DocPage title="Layout">
+    <Header as="h2">Overview</Header>
+    <p>The following components are introduced by Stardust to handle layout aspects:</p>
+    <ul>
+      <li>{links.flex} - designed for layout in one dimension, either a row or a column.</li>
+      <li>
+        {links.grid} - designed for two-dimensional layout, rows and columns at the same time.
+      </li>
+    </ul>
+    <Header as="h3">Flex vs Grid</Header>
+    <p>
+      Sometimes it might not be evident if some particular case is a case of 1D (i.e. {code('Flex')}
+      ) or 2D (i.e. {code('Grid')}) layout. As a result, quite often {code('Grid')} component is
+      used at places where it is sufficient to use {code('Flex')}.
+    </p>
+    <p>Here is the question you might ask yourself to decide on that:</p>
+    <ul>
+      <li>
+        do I only need to control the layout by row <strong>or</strong> column? – if yes, use{' '}
+        {code('Flex')}.
+      </li>
+      <li>
+        do I need to control the layout by row <strong>and</strong> column? – if yes, use{' '}
+        {code('Grid')}.
+      </li>
+    </ul>
+    <Header as="h3">Note on Segment component</Header>
```

Comment:
Actually, we should have the same note about `Box` component

Reply:
nit: We can move this section after `Flex recipes`

Reply:
### Notes on Box component
speaking of the `Box` component - it seems that the first thing we should do is rather fix its description. I've intentionally consulted about whether the `Box` component is seen now as a layout one, given the recent change of its description:

Thus, lets just make things consistent - change the description of `Box` component and put an update to the guides then - otherwise it will just create conflicting thoughts in the client's head.

### Why not put after Flex recipies
This is, actually, done intentionally. Given current usage statistics, the problems that are described in this 'Notes' section is much more serious (especially taking their conequences into account) rather than the ones introduced by overuse of, say, `Flex.Item`. Thus, it is very important to make this note earlier than later. 

Another point relates to where this paragraph semantically attributes to: it definitely should be a part of the section where Stardust layout components are enumerated, with the corresponding use-cases for each - and thus I would rather leave it here and won't introduce a separate section for it.

Reply:
We added note about `Segment`, it's cool 👍 

But, I want avoid the following thing: "_Okay, I can't use `Segment`... Oh, we have have `Box`, let's do `<Box styles={{ display: 'flex' }} />`!_" I.e. `Box` can be abused for layouts purposes, too.

Reply:
agree - then, if we agree on fixing `Box` description (which now states that it is a _layout_ component), then let me do that and introduce a dedicated section to the `Layout` guide